src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -164,6 +249,21 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler,
   /// The resources changed listeners.
   std::vector<std::function<void()>> resources_changed_listeners_;
 
+  /// AutoscalerStateService only states.
+
+  absl::flat_hash_map<NodeID, rpc::NodeState> node_states_;
```

Comment:
this might introduce extra memory overhead. we probably need unify them with existing resource state.

Reply:
I think I will try to merge into https://github.com/ray-project/ray/blob/c6d6cd9fd4d94c0c619b05a327ac0830b0d0a56d/src/ray/raylet/scheduling/cluster_resource_data.h#L476-L498

Reply:
yeah this sounds good. Also feel free to defer it into a next PR